docs/platform-specification.md

```diff
@@ -651,26 +709,48 @@ The Upload action is triggered when the user clicks on the "Upload" button on th
 [`arduino-cli upload`](commands/arduino-cli_upload.md). Arduino uses the term "upload" for the process of transferring a
 program to the Arduino board.
 
-The **upload.tool** property determines the tool to be used for upload. A specific **upload.tool** property should be
-defined for every board in boards.txt:
+The **upload.tool.<protocol_name\>** property determines the tool to be used for upload. A specific
+**upload.tool.<protocol_name\>** property should be defined for every board in boards.txt:
 
     [......]
-    uno.upload.tool=avrdude
+    uno.upload.tool.serial=avrdude
     [......]
-    leonardo.upload.tool=avrdude
+    leonardo.upload.tool.serial=avrdude
+    leonardo.upload.tool.network=arduino_ota
     [......]
 
+When the user tries to upload using a certain protocol but the board doesn't support it it will fallback to `default` if
+specified. A board can always specify a `default` protocol even if others are defined:
+
+    [......]
+    uno.upload.tool.default=avrdude
+    [......]
+    leonardo.upload.tool.default=avrdude
+    leonardo.upload.tool.network=arduino_ota
+    [......]
+
+`default` is also used when no upload address is provided by the user, this can be done only for boards that use upload
+tools with builtin port detection like `openocd`.
+
+For backward compatibility with IDE 1.8.15 and older the previous syntax is still supported:
```

Comment:
We can't know exactly which version of all products will contain this feature though, I think my also be misleading if we reference versions that are still not released. Probably we should update this when new versions are released.

Reply:
Yeah, it's really not a great situation. My concern is that the need to add the information will be forgotten. But guessing the versions risks the introduction of incorrect information.

I think that this will be very important information for platform developers using pluggable discovery because they will need to clearly document the minimum supported tool versions.